File: AN_2/SEM_2/MPP/java/lab6/Services/src/main/java/ro/mpp2024/services/IServices.java
package ro.mpp2024.services;

import java.util.List;
import ro.mpp2024.model.*;

public interface IServices {
    void login(Utilizator utilizator, IObserver client) throws Exception;

    void inscriereParticipant(Participant participant) throws Exception;

    List<ParticipantDTO> findAllParticipanti() throws Exception;

    List<Participant> cautaParticipantiDupaEchipa(String numeEchipa) throws Exception;

    Echipa findEchipaByNume(String nume) throws Exception;

    void adaugaEchipa(Echipa echipa) throws Exception;

    List<Cursa> findAllCurse() throws Exception;

    List<CursaDTO> listDtoCurse() throws Exception;

    int findNrPartByCursa(Cursa cursa) throws Exception;

    void adaugaParticipantLaCursa(ParticipantCursa participantCursa) throws Exception;

    List<ParticipantCursa> findAllParticipantCursa() throws Exception;

    void logout(Utilizator utilizator, IObserver client) throws Exception;

    Participant findByNume(String nume) throws Exception;
}

File: AN_2/SEM_2/MPP/c#/lab7/Services/mpp.services/IServices.cs
namespace Services.mpp.services;
using Model.mpp.model;

public interface IServices
{
    void Login(Utilizator utilizator, IObserver client);

    void InscriereParticipant(Participant participant);

    List<Participant> FindAllParticipanti();

    List<Participant> CautaParticipantiDupaEchipa(string numeEchipa);

    Echipa FindEchipaByNume(string nume);

    void AdaugaEchipa(Echipa echipa);

    List<Cursa> FindAllCurse();

    List<CursaDTO> ListDtoCurse();

    int FindNrPartByCursa(Cursa cursa);

    void AdaugaParticipantLaCursa(ParticipantCursa participantCursa);

    List<ParticipantCursa> FindAllParticipantCursa();
    
    void Logout(Utilizator utilizator, IObserver client);

    Participant FindByNume(string nume);
}